Assets/Scripts/Utility/RMBLayout.cs

```diff
@@ -840,47 +840,59 @@ public static bool IsRepairShop(DFLocation.BuildingTypes buildingType)
+                    GameObject go;
+                    if (!(go = MeshReplacement.ImportCustomGameobject(obj.ModelIdNum, parent, modelMatrix)))
+                    {
+                        if (!hasModelData) {
+                            Debug.LogError($"Could not load model '{obj.ModelIdNum}' in block '{blockData.Name}'");
+                            continue;
+                        } else if (combiner == null || IsCityGate(obj.ModelIdNum) || IsBulletinBoard(obj.ModelIdNum) || PlayerActivate.HasCustomActivation(obj.ModelIdNum)) {
+                            AddStandaloneModel(dfUnity, ref modelData, modelMatrix, parent);
+                        } else {
+                            combiner.Add(ref modelData, modelMatrix);
+                        }
+                    }
+
                     // Store building information for first model of record
                     // First model is main record structure, others are attachments like posts
                     // Only main structure is needed to resolve building after hit-test
-                    int buildingKey = 0;
                     if (firstModel)
                     {
-                        // Create building key for this record - considered experimental for now
-                        buildingKey = BuildingDirectory.MakeBuildingKey((byte)layoutX, (byte)layoutY, (byte)recordCount);
-
                         StaticBuilding staticBuilding = new StaticBuilding();
                         staticBuilding.modelMatrix = modelMatrix;
                         staticBuilding.recordIndex = recordCount;
-                        staticBuilding.centre = new Vector3(modelData.DFMesh.Centre.X, modelData.DFMesh.Centre.Y, modelData.DFMesh.Centre.Z) * MeshReader.GlobalScale;
-                        staticBuilding.size = new Vector3(modelData.DFMesh.Size.X, modelData.DFMesh.Size.Y, modelData.DFMesh.Size.Z) * MeshReader.GlobalScale;
+                        if (go == null || modelData.Indices != null) {
+                            staticBuilding.centre = new Vector3(modelData.DFMesh.Centre.X, modelData.DFMesh.Centre.Y, modelData.DFMesh.Centre.Z) * MeshReader.GlobalScale;
+                            staticBuilding.size = new Vector3(modelData.DFMesh.Size.X, modelData.DFMesh.Size.Y, modelData.DFMesh.Size.Z) * MeshReader.GlobalScale;
+                        } else {
+                            Renderer goRenderer = go.GetComponent<Renderer>();
+                            //staticBuilding.centre = new Vector3(goRenderer.bounds.center.x, goRenderer.bounds.center.y, goRenderer.bounds.center.z);
+                            staticBuilding.size = new Vector3(goRenderer.bounds.size.x, goRenderer.bounds.size.y, goRenderer.bounds.size.z);
+                            staticBuilding.centre = staticBuilding.size / 2;    // Seems to replicate DFMesh.Centre whereas goRenderer.bounds.center.z gives position.z for some reason.
```

Comment:
The commented out line was what I thought I should use for the centre but the z is the same as parent transform position z which is incorrect. Unless anyone objects I will remove the commented out line.

Reply:
Removed.

Reply:
Actually it turns out calculating this works fine but only if the centre of the custom model is at the exact centre which is true for DF models but not true of the test custom model. Trying to find a solution to avoid all models having to have a dead centre 0,0,0

Reply:
Solved this now.